packages/actor-rdf-parse-shaclc/lib/ActorRdfParseShaclc.ts

```diff
@@ -0,0 +1,45 @@
+import type {
+  IActionRdfParse,
+  IActorRdfParseFixedMediaTypesArgs,
+  IActorRdfParseOutput,
+} from '@comunica/bus-rdf-parse';
+import {
+  ActorRdfParseFixedMediaTypes,
+} from '@comunica/bus-rdf-parse';
+import type { IActionContext } from '@comunica/types';
+import { wrap } from 'asynciterator';
+import { parse } from 'shaclc-parse';
+
+/**
+ * A comunica SHACL Compact Syntax RDF Parse Actor.
+ */
+export class ActorRdfParseShaclc extends ActorRdfParseFixedMediaTypes {
+  /**
+   * @param args -
+   *   \ @defaultNested {{
+   *       "text/shaclc": 1.0
+   *     }} mediaTypePriorities
+   *   \ @defaultNested {{
+   *       "text/shaclc": "http://www.w3.org/ns/formats/Shaclc"
+   *     }} mediaTypeFormats
+   */
+  public constructor(args: IActorRdfParseFixedMediaTypesArgs) {
+    super(args);
+  }
+
+  public async runHandle(action: IActionRdfParse, mediaType: string, context: IActionContext):
+  Promise<IActorRdfParseOutput> {
+    return { data: <any> wrap(toString(action.data).then(str => parse(str))), metadata: { triples: true }};
```

Comment:
I think we don't strictly require `asynciterator` here, we can probably do this using regular Node streams.

Reply:
I'm not immediately sure how to achieve this with the current logic to emit prefixes in there as well; is having `asynciterator` here problematic in any way?

Reply:
> I'm not immediately sure how to achieve this with the current logic to emit prefixes in there as well

Not sure I understand. Is asynciterator doing anything with prefixes here?

> is having asynciterator here problematic in any way?

I just think that the Node-way would be a simpler way to solve the problem, as we don't need the additional logic complexity of asynciterator.

Reply:
Ok, let's keep asynciterator here, will indeed be simpler :-)